icefoganalytics/travel-authorization
# Ytgov Issue 113: Travel Desk Request Missing Fields in Grid

Fixes https://github.com/ytgov/travel-authorization/issues/113

# Context

**Describe the bug** On the https://travel-auth-dev.ynet.gov.yk.ca/travel-desk grid screen the user, department and branch fields are not being populated. **To Reproduce** Steps to reproduce the behavior:

Updated steps to reproduce.

1. In the top right click on the kebab menu and select Administration.
2. Click on the User Management entry.
3. Find your user in the table and click on them.
4. Set your user's first name, last name, and department and then click save.
5. Note that there is no way to edit your division, branch or unit via the UI.
6. In the database, find your user's record update the department, division, and branch to match something like
    `Health and Social Services -> Community and Primary Care -> Integrated Health Services`
    This information can be found in the `YgDepartments` table.
7. Go to your profile via the top nav kebab menu and refresh the page to check that your details are correct.
8. Go to the My Travel Request page via the top drop down nav.
9. Click on the "+ Travel Authorization" button to create a travel authorization.
10. Fill in the first two sections and make sure that your travel method is "Aircraft".
11. Generate the estimate via the call-to-action button.
12. Submit to yourself as the supervisor.
13. Go to the Manager View via the top drop down nav.
14. Find your approval in the Pending Approvals section and click on it.
15. Scroll down to the bottom and approve it via the button and dialog.
16. Go back to the "My Travel Request" page via the top drop down nav.
17. Find you travel request in the table and click the "Submit Travel Desk Request button.
18. Fill in all required fields and submit the request.
19. Go to the "Travel Desk" page via the top drop down nav.
20. Find the request in the Travel Desk Requests table, and note that the department and branch are missing.
    > I fixed the Name column in a [previous PR](https://github.com/icefoganalytics/travel-authorization/commit/78c50cd68b54614a57af454f34edc3db2c945cc0#diff-12d5e912bb3fcca245170cb3a8289981413425f1f621fde99f170069b2c3ce75R41).

Note that the "Travel Start Date" column is returning "Dec 29 1969" which is unlikely to be correct.
The "Travel End Date" column is blank.

**Expected behavior** The traveler name, department and branch should be populated.

**Screenshots** <img alt="image" width="944" src="https://private-user-images.githubusercontent.com/89539008/353542062-2d436649-9543-46de-adf2-d6ec11495e82.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0Nzg1NDEsIm5iZiI6MTcyNzQ3ODI0MSwicGF0aCI6Ii84OTUzOTAwOC8zNTM1NDIwNjItMmQ0MzY2NDktOTU0My00NmRlLWFkZjItZDZlYzExNDk1ZTgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDIzMDQwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlNzlmOTZmYzU5NTYyNDg3ZjIxMjM2MTJjMjk3OGZjYWZjY2M2ODM0NzQ3MzBhNmZhMGI4MDBlOWU4MTMxNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.g-3mNRq11Myc79r3DDpBxrviii1hnqfLT5bzKqQ5M9Q">

**Desktop (please complete the following information):**

- Browser: Chrome

- Version v2024.7.29.1

# Implementation

1. Standardize travel desk page component.
2. Break export to csv code out into a component.
3. Move csv and print data serialization to back-end.
4. Standardize travel-desk-travel-requests use files, front-end api, back-end api, policy and serializer.
5. Update TravelSegment#departureOn field type to match actual type returned by Sequelize. DATEONLY returns a string, but accepts both Date and string. Sadly Sequelize + Typescript doesn't support one format for setter and a different format for getter.


# Screenshots

Table View with fixed data
![image](https://github.com/user-attachments/assets/d480d394-d03b-489b-a058-99b792531750)

Pre-Print view with fixed data
![image](https://github.com/user-attachments/assets/7ff0d66e-2775-4889-968d-22de2e91094b)

Print view with fixed data
![image](https://github.com/user-attachments/assets/e07adc27-bb76-417d-8c5f-40b8dbe6bd21)

CSV export of table
![image](https://github.com/user-attachments/assets/b9152128-0cc4-4c4b-ab44-c3e3a74f2ef7)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
4. Log in to the app at http://localhost:8080
1. In the top right click on the kebab menu and select Administration.
2. Click on the User Management entry.
3. Find your user in the table and click on them.
6. Set your user's first name, last name, and department and then click save.
7. Note that there is no way to edit your division, branch or unit via the UI.
8. In the database, find your user's record update the department, division, and branch to match something like
    `Health and Social Services -> Community and Primary Care -> Integrated Health Services`
    This information can be found in the `YgDepartments` table.
9. Go to your profile via the top nav kebab menu and refresh the page to check that your details are correct.
10. Go to the My Travel Request page via the top drop down nav.
11. Click on the "+ Travel Authorization" button to create a travel authorization.
12. Fill in the first two sections and make sure that your travel method is "Aircraft".
13. Generate the estimate via the call-to-action button.
14. Submit to yourself as the supervisor.
15. Go to the Manager View via the top drop down nav.
16. Find your approval in the Pending Approvals section and click on it.
17. Scroll down to the bottom and approve it via the button and dialog.
18. Go back to the "My Travel Request" page via the top drop down nav.
19. Find you travel request in the table and click the "Submit Travel Desk Request button.
20. Fill in all required fields and submit the request.
21. Go to the "Travel Desk" page via the top drop down nav.
22. Check that the table now has the correct Department, Branch, Travel Start Date, Travel End Date info.
23. Export to CSV, check the data is correct.
24. Print the report, check the data is correct.
